### ISSUE TITLE
Modal dialog: small improvement for `elementShouldBeHidden`

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#65829]
### Description

The modal dialog component uses some logic to determine what body children should be `aria-hidden` when the modal dialog is open.

https://github.com/WordPress/gutenberg/blob/559391a6f1451e68124c9a18925ab784bca80e29/packages/components/src/modal/aria-helper.ts#L23-L54

Some elements are correctly filtered out as they should not get the `aria-hidden` attribute. It looks like elements with the `hidden` HTML attribute could be filtered out as well, as they're already completely hidden from assistive technologies.

Currentlty, `aria-hidden` is added and then removed also to these elements, which seems unnecessary:

![Image](https://github.com/user-attachments/assets/21818b5d-f35a-42cf-a5d3-9b618489a7bd)

### Step-by-step reproduction instructions

- Go to the post editor.
- Inspect the DOM.
- For the purpose of this test, make sure no `speak` messages have been sent to the aria-live regions.
- When no messages have been sent yet, observe the element with id `a11y-speak-intro-text` does have an HTML attribute `hidden`.
- Open a modal dialog e.g. the Keyboard Shortcuts one by pressing the keys Control+Option+H.
- Observe the `a11y-speak-intro-text` element gets an `aria-hidden="true"` attribute, which is unnecessary as the element is already hidden.
- There may be other elements with a `hidden` HTML attribute e.g. added by plugins that don't need `aria-hidden="true"`.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

**Comment by vk17-starlord** [comment]
The code could be optimized by checking whether an element already has the hidden attribute, and if so, skip applying aria-hidden="true". The current behavior is redundant, as hidden already takes care of making the element inaccessible to assistive technologies.

This could be achieved by modifying the elementShouldBeHidden() function to include a condition like:

javascript

if (element.hasAttribute('hidden')) {
    return false;
}

should i raise PR with this change @afercia ?

**Comment by afercia** [comment]
@vk17-starlord sure please do feel free to submit a PR.
I think the best approach would be to add one more 'OR' here, where some element with sprecific attributes are already checked. 

https://github.com/WordPress/gutenberg/blob/d6a4c72606d97d4b9e1593cce28f6bf9d5448052/packages/components/src/modal/aria-helper.ts#L46-L54

**Comment by vk17-starlord** [comment]
@afercia sure working on it

**Comment by vk17-starlord** [comment]
@afercia can you check this ?
